src/ucp/core/ucp_worker.c

```diff
@@ -1828,6 +1828,12 @@ ucp_worker_get_ep_config(ucp_worker_h worker, const ucp_ep_config_key_t *key,
     for (ep_cfg_index = 0; ep_cfg_index < worker->ep_config_count;
          ++ep_cfg_index) {
         if (ucp_ep_config_is_equal(&worker->ep_config[ep_cfg_index].key, key)) {
+            if (!(key->flags & UCP_EP_CONFIG_FLAG_FAKE)) {
+                /* If the new configuration is equal to the configuration in the
+                 * Worker and it's not a fake configuration, remove FAKE flag */
```

Comment:
/* if remote info does not affect local config, it's not a local one any more */